src/features/tokens/routes.ts

```diff
@@ -57,35 +58,36 @@ export function useTokenRoutes() {
 async function fetchRemoteTokensForCollateralToken(
   token: TokenMetadata,
 ): Promise<TokenMetadataWithHypTokens> {
-  const { chainId, symbol, decimals, hypCollateralAddress } = token;
+  const { type, chainId, symbol, decimals, hypWrapperAddress } = token;
   logger.info('Inspecting token:', symbol);
   const provider = getProvider(chainId);
-  const collateralContract = getHypErc20CollateralContract(hypCollateralAddress, provider);
-
-  logger.info('Validating token metadata');
-  const wrappedTokenAddr = await collateralContract.wrappedToken();
-  const erc20 = getErc20Contract(wrappedTokenAddr, getProvider(chainId));
-  const decimalsOnChain = await erc20.decimals();
-  if (decimals !== decimalsOnChain) {
-    throw new Error(
-      `Token config decimals ${decimals} does not match contract decimals ${decimalsOnChain}`,
-    );
-  }
-  const symbolOnChain = await erc20.symbol();
-  if (symbol !== symbolOnChain) {
-    throw new Error(
-      `Token config symbol ${symbol} does not match contract decimals ${symbolOnChain}`,
-    );
+  const wrapperContract = getHypWrapperContract(type, hypWrapperAddress, provider);
+
+  if (type === TokenType.collateral) {
```

Comment:
Is this not necessary for native as well?

Reply:
What would we validate it against?